### PR TITLE
fix(components): [el-date-picker] panel not select value enter error

### LINKED ISSUE #3767: [Bug Report] press enter report error when date-picker panel not select value
### Element Plus version
1.1.0-beta.19

### OS/Browsers version
win10/chrome

### Vue version
3.2.19

### Reproduction Link
https://github.com/element-plus/element-plus

### Steps to reproduce
1、focus the input，show the date panel
2、press enter
image

### What is Expected?
Don't report errors

### What is actually happening?
Look at the picture above

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #3767

### CODE REVIEW

File: packages/components/date-picker/src/date-picker-com/panel-date-pick.vue [hunk 1]
```diff
@@ -513,6 +513,7 @@ export default defineComponent({
     }
 
     const isValidValue = (date_) => {
```
Comment (sxzz) [inline packages/components/date-picker/src/date-picker-com/panel-date-pick.vue:515]: We should add `unknown` type, and check if `date_`(I suggest it should be renamed to `date`) is `dayjs`
```ts
date instanceof dayjs
```

Comment (btea) [reply packages/components/date-picker/src/date-picker-com/panel-date-pick.vue]: Maybe we should add `Dayjs` type ?

File: packages/components/date-picker/src/date-picker-com/panel-date-pick.vue [hunk 2]
```diff
@@ -512,9 +512,10 @@ export default defineComponent({
       }
     }
 
-    const isValidValue = (date_) => {
+    const isValidValue = (date: Dayjs) => {
+      if (!(date instanceof dayjs)) return false
       return (
-        date_.isValid() && (disabledDate ? !disabledDate(date_.toDate()) : true)
+        date.isValid() && (disabledDate ? !disabledDate(date.toDate()) : true)
       )
```
Comment (sxzz) [inline packages/components/date-picker/src/date-picker-com/panel-date-pick.vue:519]:
```ts
import { Dayjs } from 'dayjs'
const isValidValue = (date: unknown) => {
  return (
    date instanceof Dayjs &&
    date.isValid() &&
    (disabledDate ? !disabledDate(date.toDate()) : true)
  )
}
```

Comment (btea) [reply packages/components/date-picker/src/date-picker-com/panel-date-pick.vue:519]: If so, the `Dayjs` type need to remove
https://github.com/element-plus/element-plus/blob/e5025ce50deb167017b6c3b7c11eb8d3bfbb0d5a/packages/components/date-picker/src/date-picker-com/panel-date-pick.vue#L191

Comment (sxzz) [reply packages/components/date-picker/src/date-picker-com/panel-date-pick.vue:519]: Yes.

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
@btea It seems can't fix bug. Let's use `dayjs.isDayjs(dayjs())`. My fault.

**Comment by btea** [comment]
@sxzz I've fixed it.

**Review by sxzz — APPROVED** [review]

**Pull request merged**